crates/core/src/formatter.rs

```diff
@@ -862,3 +866,157 @@ Refunded: {:.10} ETH
+        out += "    |\n";
+        let doc = match self.error.get_documentation() {
+            Ok(opt) => opt,
+            Err(e) => {
+                tracing::info!("Failed to get error documentation: {}", e);
+                None
+            }
+        };
+        let summary = doc
+            .as_ref()
+            .map_or("An unknown error occurred", |d| d.summary.as_str());
+        out += &format!("    = {} {}\n", "error:".bright_red(), summary);
+        out
+    }
+
+    /// Returns the transaction details if available.
+    fn tx_details(&self) -> String {
+        let mut out = String::new();
+        if let Some(tx) = self.tx {
+            out += "    | \n";
+            out += &format!("    | {}\n", "Transaction details:".cyan());
+            out += &format!(
+                "    |   Transaction Type: {:?}\n",
+                tx.common_data.transaction_type
+            );
+            out += &format!("    |   Nonce: {}\n", tx.nonce());
+            if let Some(contract_address) = tx.recipient_account() {
+                out += &format!("    |   To: {:?}\n", contract_address);
+            }
+            out += &format!("    |   From: {:?}\n", tx.initiator_account());
+            if let Some(input_data) = &tx.common_data.input {
+                let hex_data = input_data.data.encode_hex();
+                out += &format!("    |   Input Data: 0x{}\n", hex_data);
+                out += &format!("    |   Hash: {:?}\n", tx.hash());
+            }
+            out += &format!("    |   Gas Limit: {}\n", tx.common_data.fee.gas_limit);
+            out += &format!(
+                "    |   Gas Price: {}\n",
+                format_gwei(tx.common_data.fee.max_fee_per_gas)
+            );
+            out += &format!(
+                "    |   Gas Per Pubdata Limit: {}\n",
+                tx.common_data.fee.gas_per_pubdata_limit
+            );
+
+            // Log paymaster details if available.
+            let paymaster_address = tx.common_data.paymaster_params.paymaster;
+            let paymaster_input = &tx.common_data.paymaster_params.paymaster_input;
+            if paymaster_address != Address::zero() || !paymaster_input.is_empty() {
+                out += &format!("    | {}\n", "Paymaster details:".cyan());
+                out += &format!("    |   Paymaster Address: {:?}\n", paymaster_address);
+                let paymaster_input_str = if paymaster_input.is_empty() {
+                    "None".to_string()
+                } else {
+                    paymaster_input.encode_hex()
+                };
+                out += &format!("    |   Paymaster Input: 0x{}\n", paymaster_input_str);
+            }
+        }
+        out
+    }
+
+    /// Returns documentation details including likely causes, fixes, and references.
+    fn docs(&self) -> String {
+        let mut out = String::new();
+        if let Ok(Some(doc)) = self.error.get_documentation() {
+            if !doc.likely_causes.is_empty() {
```

Comment:
Proposal for the future: given that we're working with pretty complex message templates in several places in the code, maybe it's the time when we start using [handlebars](https://docs.rs/handlebars/latest/handlebars/) for templating, as it will likely be much more maintainable.

Not proposing to do it in this PR.